crates/invoker-impl/src/lib.rs

```diff
@@ -384,12 +414,16 @@ where
                     }
                 }
             },
-
-            Some(invoke_input_command) = segmented_input_queue.dequeue(), if !segmented_input_queue.is_empty() && self.quota.is_slot_available() => {
+            true = &mut token_state => {
```

Comment:
Just testing my understanding: does it matter if this is written as `_ = &mut token_state => { ... }`? The return value is of no significance in this arm, as long as we poll the token state?

Reply:
It actually does matter. If this poll returns `true` it means that the token state has changed (call to is_ready() was false and now it should return true). Which means we can now try the entire `select!` again (since it's in a loop) because other branches are now enabled (since is_ready() now returns true).

on the other hand if this branch returns false, which means a token is ready BUT there is no change in state (is_ready() was already returning true even before the poll) which means we can safely ignore/disable this branch and just wait for another select branch to finish. If we used the pattern `_ = &mut token_state` we will just get stuck in a busy loop